volatility/framework/plugins/mac/vfsevents.py

```diff
@@ -0,0 +1,76 @@
+# This file is opyright 2020 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+
+from typing import Iterable, Callable, Tuple
+
+from volatility.framework import renderers, interfaces, constants, exceptions, contexts
+from volatility.framework.configuration import requirements
+from volatility.framework.objects import utility
+
+class VFSevents(interfaces.plugins.PluginInterface):
+    """ Lists processes that are filtering file system events """ 
+
+    event_types = ["CREATE_FILE", 
+                   "DELETE", 
+                   "STAT_CHANGED", 
+                   "RENAME", 
+                   "CONTENT_MODIFIED", 
+                   "EXCHANGE", 
+                   "FINDER_INFO_CHANGED", 
+                   "CREATE_DIR", 
+                   "CHOWN",
+                   "XATTR_MODIFIED", 
+                   "XATTR_REMOVED", 
+                   "DOCID_CREATED", 
+                   "DOCID_CHANGED"]
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.TranslationLayerRequirement(name = 'primary',
+                                                     description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(name = "darwin", description = "Mac kernel"),
+        ]
+
+    def _generator(self):
+        """
+        Lists the registered VFS event watching processes
+        Also lists which event(s) a process is registered for
+        """
+
+        kernel = contexts.Module(self.context, self.config['darwin'], self.config['primary'], 0)
+    
+        watcher_table = kernel.object_from_symbol("watcher_table")
+
+        for watcher in watcher_table:
+            if watcher == 0:
```

Comment:
Could be `if not watcher:`?  It's purely a stylistic thing and technically not identical, so totally your call.  5:)